etna/datasets/tsdataset.py

```diff
@@ -129,13 +130,30 @@ def make_future(self, future_steps: int) -> "TSDataset":
         future_ts.df_exog = self.df_exog
         return future_ts
 
-    def _merge_exog(self, df):
-        if df.index.min() < self.df_exog.index.min():
-            raise ValueError(
-                f"Exog data does not have enough history:" f"{df.index.min()} < {self.df_exog.index.min()}"
-            )
-        if df.index.max() > self.df_exog.index.max():
-            raise ValueError(f"Exog data does not have enough future:" f"{df.index.max()} > {self.df_exog.index.max()}")
+    @staticmethod
+    def _check_exog(df: pd.DataFrame, df_exog: pd.DataFrame):
+        """Check that df_exog have more timestamps than df."""
+        df_segments = df.columns.get_level_values("segment")
+        for segment in df_segments:
+            target = df[segment]["target"].dropna()
+            exog_regressor_columns = [x for x in set(df_exog[segment].columns) if x.startswith("regressor")]
+            for series in exog_regressor_columns:
+                exog_series = df_exog[segment][series].dropna()
+                if target.index.min() < exog_series.index.min():
+                    raise ValueError(
+                        f"All the regressor series should start not later than corresponding 'target'."
+                        f"Series {series} of segment {segment} have not enough history: "
+                        f"{target.index.min()} < {exog_series.index.min()}."
+                    )
```

Comment:
This way we won't show all problematic segments. It is not huge problem right now, but for convenience I'd like to see all segments contain this problem